src/map_votes.cpp

```diff
@@ -0,0 +1,543 @@
+	}
+}
+
+CON_COMMAND_CHAT_FLAGS(nominate, "Nominate a map", ADMFLAG_NONE)
+{
+	bool bIsClearingNomination = args.ArgC() < 2;
+	int iResponse = g_pMapVoteSystem->AddMapNomination(player->GetPlayerSlot(), bIsClearingNomination ? "" : args[1]);
+	if (iResponse == NominationReturnCodes::VOTE_STARTED) {
+		ClientPrint(player, HUD_PRINTTALK, CHAT_PREFIX "Could not nominate as the vote has already started.");
+	}
+	else if (iResponse == NominationReturnCodes::INVALID_INPUT) {
+		ClientPrint(player, HUD_PRINTTALK, CHAT_PREFIX "Could not nominate as the input is invalid. Usage: !nominate <map substring>");
+	}
+	else if (iResponse == NominationReturnCodes::MAP_NOT_FOUND) {
+		ClientPrint(player, HUD_PRINTTALK, CHAT_PREFIX "Could not nominate as no map matched '%s'.", args[1]);
+	}
+	else if (iResponse == NominationReturnCodes::INVALID_MAP) {
+		ClientPrint(player, HUD_PRINTTALK, CHAT_PREFIX "The map matching '%s' is not available for nomination.", args[1]);
+	}
+	else if (iResponse == NominationReturnCodes::NOMINATION_DISABLED) {
+		ClientPrint(player, HUD_PRINTTALK, CHAT_PREFIX "Nomination is currently disabled.");
+	}
+	else if (bIsClearingNomination) {
+		ClientPrint(player, HUD_PRINTTALK, CHAT_PREFIX "Your nomination was reset.");
+	}
+	else {
+		const char* sPlayerName = player->GetPlayerName();
+		const char* sMapName = g_pMapVoteSystem->GetMapName(iResponse);
+		int iNumNominations = g_pMapVoteSystem->GetTotalNominations(iResponse);
+		ClientPrintAll(HUD_PRINTTALK, CHAT_PREFIX " \x06%s \x01was nominated by %s. It now has %d nominations.", sMapName, sPlayerName, iNumNominations);
+	}
+}
+
+static int __cdecl OrderStringsLexicographically(const char* const* a, const char* const* b)
+{
+	return V_strcasecmp(*a, *b);
+}
+
+CON_COMMAND_CHAT_FLAGS(maplist, "List the maps in the server", ADMFLAG_NONE)
+{
+	ClientPrint(player, HUD_PRINTTALK, CHAT_PREFIX "The list of all maps will be shown in console.");
+	ClientPrint(player, HUD_PRINTCONSOLE, "The list of all maps is:");
+	CUtlVector<const char*> vecMapNames;
+	for (int i = 0; i < g_pMapVoteSystem->GetMapListSize(); i++) {
+		vecMapNames.AddToTail(g_pMapVoteSystem->GetMapName(i));
+	}
+	vecMapNames.Sort(OrderStringsLexicographically);
+	FOR_EACH_VEC(vecMapNames, i)
+	{
+		ClientPrint(player, HUD_PRINTCONSOLE, "- %s", vecMapNames[i]);
+	}
+}
+
+CON_COMMAND_CHAT_FLAGS(nomlist, "List the list of nominations", ADMFLAG_NONE)
+{
+	ClientPrint(player, HUD_PRINTTALK, CHAT_PREFIX "Current nominations:");
+	for (int i = 0; i < g_pMapVoteSystem->GetMapListSize(); i++) {
+		if (!g_pMapVoteSystem->IsMapIndexEnabled(i)) continue;
+		int iNumNominations = g_pMapVoteSystem->GetTotalNominations(i);
+		if (iNumNominations == 0) continue;
+		const char* sMapName = g_pMapVoteSystem->GetMapName(i);
+		ClientPrint(player, HUD_PRINTTALK, CHAT_PREFIX "- %s (%d times)\n", sMapName, iNumNominations);
+	}
+}
```

Comment:
I'm not entirely sure how best to reimplement the `bIsClearingNomination` stuff with this idea, but this seems more intuitive to me:
- Cooldowns should just be displayed in the map list instead, if a map has one
- Map list should instead be shown when !nominate has no argument, perhaps this can simultaneously still clear current nomination?

I did bring this up in Discord already, just reflecting it here.

Reply:
I would treat this as not set in stone and leave it for a follow-up PR. Part of the reason to open this one is to get other servers to test it out and figure out what's missing, including 'UX' improvements.